File: Source/Aladdin.CAPI.ANSI/CAPI/ANSI/RSA/PublicKey.cs
namespace Aladdin.CAPI.ANSI.RSA
{
	///////////////////////////////////////////////////////////////////////////
	// Открытый ключ алгоритма RSA
	///////////////////////////////////////////////////////////////////////////
	public class PublicKey : CAPI.PublicKey, IPublicKey
	{
        // параметры открытого ключа 
	    private Math.BigInteger modulus; private Math.BigInteger publicExponent;

        // конструктор
	    public PublicKey(CAPI.KeyFactory keyFactory, 
            Math.BigInteger modulus, Math.BigInteger publicExponent) : base(keyFactory)
        { 
            // сохранить переданные параметры
		    this.modulus = modulus; this.publicExponent = publicExponent;
	    }
        // параметры ключа
	    public override CAPI.IParameters Parameters 
        { 
            // параметры ключа
            get { return new Parameters(modulus.BitLength, publicExponent); }
        }
	    public virtual Math.BigInteger Modulus        { get { return modulus;		 }}
	    public virtual Math.BigInteger PublicExponent { get { return publicExponent; }}
	}
}

File: Source/Aladdin.CAPI.ANSI.Java/src/aladdin/capi/ansi/rsa/PublicKey.java
package aladdin.capi.ansi.rsa;
import java.math.*; 

///////////////////////////////////////////////////////////////////////////
// Открытый ключ алгоритма RSA
///////////////////////////////////////////////////////////////////////////
public class PublicKey extends aladdin.capi.PublicKey implements IPublicKey
{
    // номер версии для сериализации
    private static final long serialVersionUID = -6578367713941521342L;

    // параметры открытого ключа 
	private final BigInteger modulus; private final BigInteger publicExponent;

    // конструктор
	public PublicKey(aladdin.capi.KeyFactory keyFactory, BigInteger modulus, BigInteger publicExponent) 
    { 
        // сохранить переданные параметры
		super(keyFactory); this.modulus = modulus; this.publicExponent = publicExponent;
	}
    // параметры ключа
	@Override public final aladdin.capi.IParameters parameters() 
    { 
        // параметры ключа
        return new Parameters(modulus.bitLength(), publicExponent); 
    }
    // параметры открытого ключа 
	@Override public final BigInteger getModulus       () { return modulus;		   }
	@Override public final BigInteger getPublicExponent() { return publicExponent; }
}
